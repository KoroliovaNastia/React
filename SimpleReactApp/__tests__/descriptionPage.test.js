import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import DescriptionPage, {mapDispatchToProps} from '../src/components/descriptionPage';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import { findNodeModule } from 'jest-resolve';
import { JestEnvironment } from '@jest/environment';
import JestMock from 'jest-mock';
import {Provider} from "react-redux";
import thunk from 'redux-thunk'
import {BrowserRouter as Router} from 'react-router-dom';


configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
let wrapper, store, match;
beforeEach(() => {
    const initialState = {
        movieState: {
            movieList: [{id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genres:['Adventure'], release_date: '2022', vote_average: 5, runtime: 154, description:""}, 
            {id: '2', image: '../src/images/movie-default.jpg', title:'Soon...', genres:['Horror'], release_date: '2020', vote_average: 4, runtime: 154, description:""}],
            movieId: 1,
            movie: {id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genres:['Adventure'], release_date: '2022', vote_average: 5, runtime: 154, description:""}
        },
        appState: {
            searchButtonText : "Search",
            logo: 'netflixroulete'
        }
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();

    match = { params: { id: 1 } }
    // wrapper = shallow(
    //     <DescriptionPage store={store} match={match}/>
    // ).dive();
});

describe('DescriptionPage component', () => {

    it('should render correctly', () => {
        wrapper = mount(<Provider store={store}><Router><DescriptionPage match={match}/></Router></Provider>)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it('should get movie and movieList correctly', () => {
        wrapper = shallow(
            <DescriptionPage store={store} match={match}/>
        ).dive();

        expect(wrapper.props().movie).toEqual({id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genres:['Adventure'], release_date: '2022', vote_average: 5, runtime: 154, description:""})
        expect(wrapper.props().movieList).toEqual([{id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genres:['Adventure'], release_date: '2022', vote_average: 5, runtime: 154, description:""}, 
        {id: '2', image: '../src/images/movie-default.jpg', title:'Soon...', genres:['Horror'], release_date: '2020', vote_average: 4, runtime: 154, description:""}])
    })
})