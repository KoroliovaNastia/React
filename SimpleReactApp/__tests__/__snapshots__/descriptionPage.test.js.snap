// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionPage component should render correctly 1`] = `
<DescriptionPage
  logo="netflixroulete"
  movieId="1"
  movieList={
    Array [
      Object {
        "description": "lorem ipsum lorem ipsum",
        "duration": "154",
        "genre": "Adventure",
        "id": "1",
        "image": "../src/images/movie-default.jpg",
        "isShowing": true,
        "rating": "5.0",
        "release_date": "2022",
        "title": "Not yet...",
      },
      Object {
        "description": "lorem ipsum lorem ipsum",
        "duration": "154",
        "genre": "Horror",
        "id": "2",
        "image": "../src/images/movie-default.jpg",
        "isShowing": true,
        "rating": "4.1",
        "release_date": "2020",
        "title": "Soon...",
      },
    ]
  }
  setIsShowing={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "description": "lorem ipsum lorem ipsum",
            "duration": "154",
            "genre": "Horror",
            "id": "2",
            "image": "../src/images/movie-default.jpg",
            "isShowing": true,
            "rating": "4.1",
            "release_date": "2020",
            "title": "Soon...",
          },
          false,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Object {
            "description": "lorem ipsum lorem ipsum",
            "duration": "154",
            "genre": "Horror",
            "id": "2",
            "image": "../src/images/movie-default.jpg",
            "isShowing": false,
            "rating": "4.1",
            "release_date": "2020",
            "title": "Soon...",
          },
        },
      ],
    }
  }
>
  <Header>
    <div
      className="uk-height-medium uk-flex uk-flex-center uk-flex-middle uk-background-cover uk-light"
      style={
        Object {
          "backgroundColor": "#2323237f",
          "backgroundImage": "url(test-file-stub)",
          "minWidth": "1200px",
        }
      }
      uk-img="true"
    >
      <Logo
        logo="netflixroulete"
      >
        <div
          className="logo"
        >
          <p>
            netflixroulete
          </p>
        </div>
      </Logo>
      <button />
      <MovieDescription
        movie={
          Object {
            "description": "lorem ipsum lorem ipsum",
            "duration": "154",
            "genre": "Adventure",
            "id": "1",
            "image": "../src/images/movie-default.jpg",
            "isShowing": true,
            "rating": "5.0",
            "release_date": "2022",
            "title": "Not yet...",
          }
        }
      >
        <Image
          image="../src/images/movie-default.jpg"
        >
          <img
            src="../src/images/movie-default.jpg"
          />
        </Image>
        <div>
          <p>
            Not yet...
          </p>
          <p>
            5.0
          </p>
        </div>
        <div>
          <p>
             year 
            154
             min
          </p>
          <p>
            lorem ipsum lorem ipsum
          </p>
        </div>
      </MovieDescription>
    </div>
  </Header>
  <Box>
    <div
      style={
        Object {
          "backgroundColor": "#555555",
          "height": "65px",
        }
      }
    >
      <p>
        Films by 
        Adventure
         genre
      </p>
    </div>
  </Box>
  <Main
    movieList={
      Array [
        Object {
          "description": "lorem ipsum lorem ipsum",
          "duration": "154",
          "genre": "Horror",
          "id": "2",
          "image": "../src/images/movie-default.jpg",
          "isShowing": false,
          "rating": "4.1",
          "release_date": "2020",
          "title": "Soon...",
        },
      ]
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      />
    </div>
  </Main>
  <Footer>
    <footer>
      <Box>
        <div
          style={
            Object {
              "backgroundColor": "#555555",
              "height": "65px",
            }
          }
        >
          <Logo
            logo="netflixroulete"
          >
            <div
              className="logo"
            >
              <p>
                netflixroulete
              </p>
            </div>
          </Logo>
        </div>
      </Box>
    </footer>
  </Footer>
</DescriptionPage>
`;
